### ISSUE TITLE
child `method() -> t` incompatible with parent `method() -> Optional[t]`

### ISSUE THREAD

**Issue by akdor1154** [python/mypy#2874]
```py
from typing import Optional, Dict
class Daddy:
        def doSomething(self) -> Optional[Dict]:
                return None

class Child(Daddy):
        def doSomething(self) -> None:
                return None
```

```sh
error: Return type of "doSomething" incompatible with supertype "Daddy"
```

Expected result: no error.
Tested against `master`, no command line parameters given.

**Comment by ilevkivskyi** [comment]
You should use ``--strict-optional`` flag, if you want mypy to carefully check ``None``s. There is a plan to turn on this option by default in one of future mypy releases.

**Comment by akdor1154** [comment]
`--strict-optional` does make the error go away. But I don't think this error is expected behaviour without `--strict-optional` - if None can satisfy Dict, then why should MyPy throw an error here?

**Comment by ilevkivskyi** [comment]
Current behaviour w.r.t. ``None`` is strange in general (moreover I think exactly this was already reported, but I could not find the issue). As I understand, there are no plans to fix this, since there will be no such thing as ``--nonstrict-optional``.

**Comment by akdor1154** [comment]
ok!
